Add support for cluster validations in gRPC clients (#10788)
grafana/mimir: docs/sources/mimir/configure/configuration-parameters/index.md

  # system as object storage backend.
  # The CLI flags prefix for this block configuration is: common.storage
  [filesystem: <filesystem_storage_backend>]

client_cluster_validation:

>   # (experimental) Optionally define client's cluster validation label.

Reviewer: Could we say `a client's` or `the client's` (depending on which makes more sense)?

Reviewer 2: How about "Optionally define the cluster validation label to be sent by the client." or something like that, involving the verb "send", as it's unclear what the client (which client btw?) is going to do with the label?

Author: I have updated the description of this cli flag in https://github.com/grafana/dskit/pull/657/commits/fc58c4cc357f5fc0327088dfe92c15ca3234b2da

[http_path_prefix: <string> | default = ""]

cluster_validation:
  # (experimental) Optionally define the server's cluster validation label. When

>   # the validation is enabled, this value will be compared with the cluster

Reviewer: ```suggestion
  # the validation is enabled, this value is compared to the cluster
```

  # CLI flag: -alertmanager.alertmanager-client.connect-backoff-max-delay
  [connect_backoff_max_delay: <duration> | default = 5s]

  cluster_validation:

>     # (experimental) Optionally define client's cluster validation label.

Reviewer: Same comment as above.

    # CLI flag: -alertmanager.alertmanager-client.cluster-validation.label
    [label: <string> | default = ""]

>   # (experimental) Optionally define gRPC client's cluster validation label.

Reviewer: Same, is there more than one gRPC client? Is yes, we could say `a gRPC client`. If no, we could say `the gRPC client`.

```yaml
# Configures the gRPC client used to communicate with ingesters from
# distributors, queriers and rulers.
grpc_client_config:

>   # (advanced) gRPC client max receive message size (bytes).

Reviewer 2: Hum why did this change? I also don't see the removal of `<grpc_client>` section, so is it repeated now?

Reviewer 2: Oh, I know why, it's because you wrapped the config with another type. If you remove the wrapping (see the other comment about just moving `RegisteredFlags` to dskit) this will be reverted.

# if ConnectTimeout > 0.
# CLI flag: -<prefix>.connect-backoff-max-delay
[connect_backoff_max_delay: <duration> | default = 5s]

cluster_validation:

>   # (experimental) Optionally define client's cluster validation label.

Reviewer: Same as above.

  # CLI flag: -<prefix>.cluster-validation.label
  [label: <string> | default = ""]

> # (experimental) Optionally define gRPC client's cluster validation label.

Reviewer: Same here.
